fairseq2n/src/fairseq2n/data/multi_data_source.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <vector>
+
+#include "fairseq2n/data/data_pipeline.h"
+
+namespace fairseq2n::detail {
+
+class multi_data_source final : public data_source {
```

Comment:
It is subjective, but I think `composite_data_source` would sound more accurate. "Composition" has a more clear meaning in software engineering. `multi` sounds a bit ambiguous to me.